msteams-platform/m365-apps/combine-office-add-in-and-teams-app.md

```diff
@@ -0,0 +1,673 @@
+
+    ```
+    |-- .vscode/
+    |-- add-in/
+    |-- |-- appPackage/assets/
+    |-- |-- infra/
+    |-- |-- src/
+    |-- |-- .eslintrc.json
+    |-- |-- babel.config.json
+    |-- |-- package-lock.json
+    |-- |-- package.json
+    |-- |-- tsconfig.json
+    |-- |-- webpack.config.js
+    |-- appPackage/
+    |-- build\appPackage/
+    |-- env/
+    |-- infra/
+    |-- node_modules/
+    |-- tab/
+    |-- |-- infra/
+    |-- |-- node_modules/
+    |-- |-- src/
+    |-- |-- .webappignore
+    |-- |-- package-lock.json
+    |-- |-- package.json
+    |-- |-- tsconfig.json
+    |-- |-- web.config
+    |-- .gitignore
+    |-- .localConfigs
+    |-- package.json
+    |-- teamsapp.local.yml
+    |-- teamsapp.yml
+    ```
+
+### Edit the tooling configuration files
+
+1. Open the `package.json` file in the root of the project.
+1. Add the following scripts to the "scripts" object:
+
+    ```
+    "install:add-in": "cd add-in && npm install",
+    "postinstall": "npm run install:add-in && npm run install:tab",
+    "build:add-in": "cd add-in && npm run build",
+    "build:add-in:dev": "cd add-in && npm run build:dev",
+    "build": "npm run build:tab && npm run build:add-in",
+    ```
+
+1. Open the `package.json` file in the **add-in** folder (not the **tab** folder or the root of the project). 
+1. Three of the scripts in the `scripts` object have a `manifest.json` parameter as in the following examples: 
+
+    ```
+    "start": "office-addin-debugging start appPackage/manifest.json",
+    "stop": "office-addin-debugging stop appPackage/manifest.json",
+    "validate": "office-addin-manifest validate appPackage/manifest.json",
+    ```
+
+    In the `start` script, change `appPackage/manifest.json` to `../appPackage/build/appPackage.local.zip`. When you're done, the line must look like this:
+
+    ```
+    "start": "office-addin-debugging start ../appPackage/build/appPackage.local.zip",
+    ```
+
+    In the `validate` and `stop` scripts, change the parameter to `../appPackage/build/manifest.local.json`. When you're done, the updates must look like this:
+
+    ```
+    "stop": "office-addin-debugging stop ../appPackage/build/manifest.local.json",
+    "validate": "office-addin-manifest validate ../appPackage/build/manifest.local.json",
+    ```
+
+1. In Visual Studio Code, open **TERMINAL**. 
+1. Navigate to the **add-in** folder and run the command `npm install`. 
+1. In the **add-in** folder, open the `webpack.config.js` file. 
+1. Change the line `from: "appPackage/manifest*.json",` to `from: "../appPackage/build/manifest*.json",`.
+1. In the root of the project, open the `teamsapp.local.yml` file and find the `provision` section. Use the `#` character to comment out the lines that validate the manifest template. This is necessary because the Teams manifest validation system is not yet compatible with the changes you made to the manifest template. When you're done, the lines should look like the following code:
```

Comment:
```suggestion
1. In the root of the project, open the `teamsapp.local.yml` file and find the `provision` section. Use the `#` character to comment out the lines that validate the manifest template. This is necessary because the Teams manifest validation system isn't yet compatible with the changes you made to the manifest template. When you're done, the lines should look like the following code:
```